python/cudf/cudf_pandas_tests/test_cudf_pandas.py

```diff
@@ -1927,3 +1927,38 @@ def test_series_dtype_property():
     expected = np.dtype(s)
     actual = np.dtype(xs)
     assert expected == actual
+
+
+def test_cudf_series_from_cudf_pandas():
+    s = xpd.Series([1, 2, 3])
+    gs = cudf.Series(s)
```

Comment:
This line would have worked even without the changes in this PR, right? It would just have been extra slow because we would have turned the Series into a pandas Series and then back into a cudf Series? Is there a way that we can test that? For example, monkey-patch `pd.Series.__init__` to always raise then run this test? pytest-xdist is multi-process, not multithreaded, so we shouldn't have to worry about locking to avoid breaking other tests.

We could also do something slightly less direct like monkey-patching `as_gpu_object` to ensure that it's called. That doesn't feel like as solid of a test, though.

Reply:
Hmm, let me try this out..

Reply:
Nah that won't be a robust test either. We will either have to time it or cprofile the call and validate the call-stack. Timing is not so stable on CI runs. I'll work on adding a profiled pytest but will do so in a later PR (for 24.04).

Reply:
@vyasr I have added more robust testing that will validate what we are calling.